src/term/transform/linearize.rs

```diff
@@ -255,25 +252,56 @@ fn get_var_uses(nam: Option<&Name>, var_uses: &HashMap<Name, Val>) -> Val {
   if let Some(nam) = nam { *var_uses.get(nam).unwrap() } else { 0 }
 }
 
-fn make_dup_tree(nam: &Name, nxt: &mut Term, uses: Val, dup_body: Option<&mut Term>) {
-  // TODO: Is there a difference between a list of dups and a complete binary tree of dups
-  // Creates this: "dup x1 x1_dup = body; dup x2 x2_dup = x1_dup; dup x3 x4 = x2_dup; nxt"
+// TODO: Is it really necessary to `count_var_uses_in_term` before `term_to_affine`?
+// Can't it be unified in the same function? (given that the actual var uses in a term is already dynamic)
+//
+/// When a term is removed, the number of uses of a variable can change inside the term_to_affine call.
+/// This returns the updated count instead of using the one we calculated initially
+fn actual_var_use(var_uses: &HashMap<Name, Val>, name: &Name) -> Val {
+  var_uses.get(&used_var_counter(name)).copied().unwrap_or(0)
```

Comment:
Wouldn't it be better to use the `entry().or_default()` pattern we used in the other passes and in some other places here?